src/stdlib_io.fypp

```diff
@@ -143,15 +145,52 @@ contains
         read(s, *)
       end do
 
-      do i = 1, max_rows_
-        #:if 'real' in t1
-          read(s, "(*"//FMT_REAL_${k1}$(1:len(FMT_REAL_${k1}$)-1)//",1x))") d(i, :)
-        #:elif 'complex' in t1
-          read(s, "(*"//FMT_COMPLEX_${k1}$(1:len(FMT_COMPLEX_${k1}$)-1)//",1x))") d(i, :)
-        #:else
-          read(s, *) d(i, :)
-        #:endif
-      end do
+      #:if 'real' in t1
+      ! Default to format used for savetxt if fmt not specified.
+      fmt_ = optval(fmt, "(*"//FMT_REAL_${k1}$(1:len(FMT_REAL_${k1}$)-1)//",1x))")
+
+      if ( fmt_ == '*' ) then
+        ! Use list directed read if user has specified fmt='*'
+        do i = 1, max_rows_
+          read (s,*) d(i, :)
+        enddo
+      else
+        ! Otherwise pass default or user specified fmt string.  
+        do i = 1, max_rows_
+          read (s,fmt_) d(i, :)
+        enddo
+      endif
```

Comment:
```suggestion
        ! Use list directed read if user has specified fmt='*'
        ! Otherwise pass default or user specified fmt string.  
        do i = 1, max_rows_
          read (s,fmt_) d(i, :)
        enddo
```

This should worl too, right?

Reply:
No, sadly that doesn't work.   * is not equal to "*"

Discussed in more detail here.  
https://fortran-lang.discourse.group/t/format-string-corresponding-to-list-directed-i-o/522/10

Reply:
Ok. Thank you for the explanation!